modules/serverbidBidAdapter.js

```diff
@@ -125,84 +79,114 @@ ServerBidAdapter = function ServerBidAdapter() {
       if (placement.networkId && placement.siteId) {
         data.placements.push(placement);
       }
-    }
+    });
 
-    if (data.placements.length) {
-      ajax(config.BASE_URI, _responseCallback, JSON.stringify(data), { method: 'POST', withCredentials: true, contentType: 'application/json' });
-    }
-  }
+    ret.data = JSON.stringify(data);
+    ret.bidRequest = validBidRequests;
+    ret.url = ENDPOINT_URL;
 
-  function _responseCallback(result) {
+    return ret;
+  },
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse, bidRequest) {
     let bid;
+    let bids;
     let bidId;
     let bidObj;
-    let bidCode;
-    let placementCode;
+    let bidResponses = [];
 
-    try {
-      result = JSON.parse(result);
-    } catch (error) {
-      utils.logError(error);
-    }
+    bids = bidRequest.bidRequest;
 
-    for (let i = 0; i < bidIds.length; i++) {
-      bidId = bidIds[i];
-      bidObj = utils.getBidRequest(bidId);
-      bidCode = bidObj.bidder;
-      placementCode = bidObj.placementCode;
+    for (let i = 0; i < bids.length; i++) {
+      bid = {};
+      bidObj = bids[i];
+      bidId = bidObj.bidId;
+      bid.bidderCode = bidObj.bidder;
```

Comment:
`bidderCode` will be set automatically by `bidderFactory` now, this line can be dropped

Reply:
Can you show me where this is the case in the code because removing that line causes tests around bidderCode to fail.